### PR TITLE
feat: check available tools in new command

### PR DESCRIPTION
**Pull request by sigmaSd**
fix https://github.com/denoland/wasmbuild/issues/48

### CODE REVIEW

**Comment by sigmaSd** [comment]
side note: maybe deno should make console.warn and console.error more special , maybe some color changes or prefix them with warning/error emoji

File: lib/commands/new_command.ts [hunk 1]
```diff
@@ -116,3 +118,40 @@ async function getFileTextIfExists(path: string) {
     }
   }
 }
+
+async function checkIfRequiredToolsExist() {
+  const requiredTools = ["deno", "cargo", "rustup"];
+  const optionalTools = ["wasm-bindgen"];
```
Comment (dsherret) [inline lib/commands/new_command.ts:124]: We can get rid of this. `wasm-bindgen` is not necessary for `wasmbuild`.

**Review by dsherret — CHANGES_REQUESTED** [review]
Almost there!

File: lib/commands/new_command.ts (hunk 1 above)
Comment (sigmaSd) [reply lib/commands/new_command.ts:124]: (this part= the optional tool part not the code I linked)

Comment (dsherret) [reply lib/commands/new_command.ts:124]: That’s only for when wasmbuild is building itself. It’s not used by consumers of wasmbuild.

Comment (sigmaSd) [reply lib/commands/new_command.ts]: make sense, I updated the code

File: lib/commands/new_command.ts [hunk 2]
```diff
@@ -116,3 +118,28 @@ async function getFileTextIfExists(path: string) {
...
+    } catch (e) {
+      if (e instanceof Deno.errors.NotFound) {
+        notInstalled.push(tool);
+      }
+    }
+  }
+
+  if (notInstalled.length !== 0) {
+    throw "Some required tools are missing: " + notInstalled;
+  }
```
Comment (dsherret) [inline lib/commands/new_command.ts:144]:
```suggestion
  if (notInstalled.length > 0) {
    throw new Error("Some required tools are missing: " + notInstalled.join(", "));
  }
```

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

File: lib/commands/new_command.ts [hunk 3]
```diff
@@ -116,3 +118,28 @@ async function getFileTextIfExists(path: string) {
...
+        .status;
+    } catch (e) {
+      if (e instanceof Deno.errors.NotFound) {
+        notInstalled.push(tool);
+      }
+    }
+  }
+
+  if (notInstalled.length > 0) {
+    throw new Error("Some required tools are missing: " + notInstalled.join(", "));
```
Comment (dsherret) [inline lib/commands/new_command.ts:143]:
```suggestion
    throw new Error(
      "Some required tools are missing: " + notInstalled.join(", "),
    );
```

**Pull request merged**